### DISCUSSION TITLE
Need help to fix an issue with antd Table version 3.x.x

### DISCUSSION THREAD

**Discussion by ananduSiva** [ant-design/ant-design#50815] [Q&A] [answered]
Simple explanation of what I am trying to do : 
I have a multiselect in table to select country list but I need to show it with "+ N more" when select is not active. The first column is fixed. My problem is when blurred the content of select is changed but the fixed column is not changing the row height accordingly. The height of original row changes automatically when the select component shrinks.

Initial:
![image](https://github.com/user-attachments/assets/c1f98e80-ef11-485e-8a5d-9b93ff87c4ba)

Selecting Countries:
![image](https://github.com/user-attachments/assets/630d4c2b-6bda-488d-a6a2-e41369190653)

onBlur:
![image](https://github.com/user-attachments/assets/883dee5e-b486-4700-91cd-efe17732ce95)

**Comment by Wxh16144** [comment]
Can you provide a snippet of code? or try add `maxTagCount: 'responsive'` to `<Select/>`

**Reply by IsKaros** [reply]
他用的3.x版本，好像`maxTagCount`没有`responsive`的选项。

**Reply by ananduSiva** [reply]
`<Table
  className={styles.tableStyles}
  scroll={{ x: 2500, y: '35vh' }}
  dataSource={cloneDeep(exclusionData)}
  columns={columns}
  pagination={paginationConfig}
  align="left"
/>

// inside the columns configuration render for the country select component
<div>
  <Select
    showSearch
    virtual
    notFoundContent={null}
    onSearch={(query) => {
      // search
    }}
    value={
      // condition to check if select is focused or not
      // if focused > values
      // if not > [values[0], "+ N more"]
    }
    filterOption={false}
    mode="multiple"
    onDropdownVisibleChange={(visible) => {
      handleDropdownVisibleChange(visible);
    }}
  />
  <Dropdown
    overlay={
      <div
        style={{
          overflow: "auto",
          transform: "translateZ(0px)",
        }}
      >
        <ul role="listbox" tabIndex={0}>
          <li
            key="All"
            role="option"
            aria-selected="false"
            onMouseDown={(e) => e.preventDefault()}
          >
            <Checkbox>All</Checkbox>
          </li>
          )}
          {visibleCountries.map((country, ind) => (
            <li role="option" aria-selected="false">
              <Checkbox>
                <span>
                  {country.code} - {country.name}
                </span>
              </Checkbox>
            </li>
          ))}
        </ul>
      </div>
    }
    visible={isVisible}
    trigger={[]}
    overlayClassName="ant-select-dropdown ant-select-dropdown--multiple ant-select-dropdown--empty"
    placement="bottomLeft"
  >
    <div />
  </Dropdown>
</div>;`

**Reply by ananduSiva** [reply]
I can make the maxTagCount work only if someone can answer these 2 questions;
![image](https://github.com/user-attachments/assets/98813460-4e17-4081-84ed-a336b584279d)

1. is there any way to change the text '+2 ...' to '+2 More'.
2. is there a way to see the selected values when focused instead of this '+2 ...'.

Thanks in advance

**Reply by IsKaros** [reply]
you can use the prop `maxTagPlaceholder` to change the text.
 In my opinion, the pic you show has already met your  need, but it seems that the style is a bit not that promising.BTW,would you mind providing a reproduce link ? I am sorry that i cannot reproduce with the code snippet above.

**Reply by ananduSiva** [reply]
I'll try to create a reproduce link for you. Meanwhile will you be able to guide me through the best way to approach this scenario? I am sure the way I did it is not optimal. how do I switch between showing everything selected and maxTagCount = 2 based on whether the select dropdown is open or not? I would really appreciate if you could provide some working example or demo.

**Reply by ananduSiva** [reply]
@IsKaros I was able to reproduce this issue this code. please let me know if you are able to access;
https://codepen.io/anandusiva/pen/vYqMYpe?editors=001

Steps to reproduce:
1. Select more than 3 options from the column 1 dropdown.
2. press Tab or click outside to blur.

Observation:
The style of fixed column is broken as it is not adjusting along with the original table row.
![image](https://github.com/user-attachments/assets/c01bd0ff-29fb-4a7f-b942-d75e384fccba)

**Reply by IsKaros** [reply]
you are right.It works when not using `fixed` prop。It can't be reproduced when using antd 4.x / 5.x。

**Reply by ananduSiva** [reply]
Is there any way you think I can fix this ? any workaround?

**Reply by ananduSiva** [reply]
@IsKaros I was able to fix this by updating the Table component's `key` whenever the onBlur function is called. Thank you for suggesting the `maxTagPlaceholder` prop. That helped me to remove some unnecessary logic to create the '+N More' link in the Select. I am closing this discussion.

**Reply by IsKaros** [reply]
Congratulations!I am glad that I can help you a bit.

**Comment by ananduSiva** [comment, marked as answer]
This issue is only seen in version 3.x.x of antd select and not in 4.x.x ro 5.x.x. The issue seems to be the fixed column table is not updating the styles when the height of the row is changed programmatically. Could be a state update issue. The solution worked for me is to udpate the `key` property of the table whenever the row height is adjusted. It will force rerender the table thus fixing the height issue for fixed column rows.
